### PR TITLE
overthebox: Add lan_if for dhcp static lease

### PR DESCRIPTION
**Pull request by angt**
Signed-off-by: Adrien Gallouët <adrien.gallouet@corp.ovh.com>

### CODE REVIEW

File: overthebox/files/etc/uci-defaults/otb-network.defaults [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+# vim: set noexpandtab tabstop=4 shiftwidth=4 softtabstop=4 :
+
+uci -q batch <<EOF
+set network.lan.proto=static
+set network.lan.ipaddr=192.168.100.1
```
Comment (gregdel) [inline overthebox/files/etc/uci-defaults/otb-network.defaults]: Not sure if that's safe, upgrade from v1 to v2 will be very tricky !

File: overthebox/files/etc/udhcpc.user [hunk 2]
```diff
@@ -37,60 +37,46 @@ set dhcpdiscovery.$INTERFACE.serverid=$serverid
...
-	/etc/init.d/firewall reload
+	set dhcp.if${_id}_gw=tag
+	add_list dhcp.if${_id}_gw.dhcp_option=1,$subnet
+	add_list dhcp.if${_id}_gw.dhcp_option=3,$router
+	add_list dhcp.if${_id}_gw.dhcp_option=6,$dns
+	commit dhcp
+	EOF
+else
+	uci -q batch <<-EOF
+	set network.lan_if${_id}.proto=static
```
Comment (gregdel) [inline overthebox/files/etc/udhcpc.user]: Missing line:
```
set network.lan_if${_id}=interface
```

File: overthebox/files/etc/udhcpc.user [hunk 3]
```diff
@@ -37,60 +37,46 @@ set dhcpdiscovery.$INTERFACE.serverid=$serverid
...
+	set network.lan_if${_id}.ipaddr=$ip
+	set network.lan_if${_id}.netmask=$subnet
+	set network.if0_dev.macaddr=auto.lan
```
Comment (gregdel) [inline overthebox/files/etc/udhcpc.user]: This does not trigger a dhcp request, it does nothing...
Changing back this line to ```set network.if0_dev.macaddr=auto.if$_id``` seems to work

**Comment by gregdel** [comment]
```
root@OverTheBox:/# ip link show if1
9: if1@eth0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1
    link/ether ca:48:7a:60:e9:b8 brd ff:ff:ff:ff:ff:ff
```

if seems down after setting up the network

File: overthebox/files/etc/uci-defaults/otb-network.defaults (hunk 1 above)
Comment (angt) [inline overthebox/files/etc/uci-defaults/otb-network.defaults]: Yep that was only a fix for the current master,
We'll do a specific PR for the v1 migration.

**Pull request merged**